packages/strapi-design-system/src/DismissableLayer/DimissableLayer.js

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+export const DismissableLayer = ({ children, className, onEscapeKeyDown, onPointerDownOutside }) => {
+  const layerRef = React.useRef(null);
+  const onEscapeKeyDownHandler = useCallbackRef(onEscapeKeyDown);
+  const onPointerDownOutsideHandler = useCallbackRef(onPointerDownOutside);
+
+  React.useEffect(() => {
+    /**
+     * @type {import('react').KeyboardEventHandler<Document>}
+     */
+    const handleKeyDown = (event) => {
+      if (event.key === 'Escape') {
+        onEscapeKeyDownHandler(event);
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, [onEscapeKeyDownHandler]);
+
+  React.useEffect(() => {
+    /**
+     * @type {import('react').PointerEventHandler<Document>}
+     */
+    const handlePointerDownOutside = (event) => {
+      if (layerRef.current && !layerRef.current.contains(event.target)) {
+        onPointerDownOutsideHandler();
+      }
+    };
+
+    document.addEventListener('pointerdown', handlePointerDownOutside);
+
+    return () => document.removeEventListener('pointerdown', handlePointerDownOutside);
+  }, [onPointerDownOutsideHandler]);
+
+  return (
+    <div ref={layerRef} className={className}>
+      {children}
+    </div>
+  );
+};
+
+DismissableLayer.propTypes = {
+  children: PropTypes.node,
+  className: PropTypes.string,
+  onEscapeKeyDown: PropTypes.func,
+  onPointerDownOutside: PropTypes.func,
```

Comment:
Are these required for the component to work?

Reply:
The component will not crash if you don't pass them. But equally the behaviour will not be what you expect if you don't pass them, additionally we can't check that you even do what we expect (close the layer) with those functions, so imo as long as the component doesn't crash we're all good.

Reply:
Thanks. If the component behaves unexpectedly, let's make these required and strict (we can open it anytime) and avoid the cost of having them optional. Wdyt?